File: Binary-Search.py
def binary_search(arr, target):
    left, right = 0, len(arr) - 1

    while left <= right:
        mid = (left + right) // 2
        if arr[mid] == target:
            return mid  # Element found
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1

    return -1  # Element not found

# Example usage
sorted_arr = [2, 3, 4, 5, 8]
target = 5
print(binary_search(sorted_arr, target))  # Output: 3

File: Binary-Search.js
function binarySearch(arr, target) {
  let left = 0;
  let right = arr.length - 1;

  while (left <= right) {
    const mid = Math.floor((left + right) / 2);
    if (arr[mid] === target) {
      return mid; // Element found
    } else if (arr[mid] < target) {
      left = mid + 1;
    } else {
      right = mid - 1;
    }
  }

  return -1; // Element not found
}

// Example usage
const sortedArr = [2, 3, 4, 5, 8];
const target = 5;
console.log(binarySearch(sortedArr, target)); // Output: 3
